### PR TITLE
docs: book pass + rpc chapter

### PR DESCRIPTION
**Pull request by onbjerg**
Documents the remaining RPC namespaces and cleans up some other parts of the book

Cleanups

- Some formatting of the book cover
- Updating the installation instructions to be up to date (GHCR over Dockerhub, no rethup, add missing platforms)

Remaining

- [x] Document every method in `eth_`
- [x] Document every method in `net_`
- [x] Document every method in `txpool_`
- [x] Document every method in `debug_`
- [x] Document every method in `trace_`
- [x] Document every method in `rpc_`

### CODE REVIEW

File: book/installation/installation.md [hunk 1]
```diff
@@ -3,13 +3,23 @@
...
 ## Hardware Requirements
 
-The hardware requirements for running Reth depend on the node configuration and can change over time as the network grows or new features are implemented. The most important requirement is by far the disk, whereas CPU and RAM requirements are relatively flexible.
+The hardware requirements for running Reth depend on the node configuration and can change over time as the network grows or new features are implemented.
+
+The most important requirement is by far the disk, whereas CPU and RAM requirements are relatively flexible.
+
+|           | Archive Node                       | Full Node                           |
+|-----------|------------------------------------|-------------------------------------|
+| Disk      | At least 2.5TB (NVMe recommended)  | TBD                                 |
```
Comment (gakonst) [inline book/installation/installation.md:19]:
```suggestion
| Disk      | At least 2 (NVMe recommended)  | TBD                                 |
```

File: book/installation/installation.md [hunk 2]
```diff
@@ -18,19 +28,21 @@ There are multiple types of disks to sync Reth, with varying size requirements,
 * Archive Node: At least 2.5TB is required to store
```
Comment (gakonst) [inline book/installation/installation.md:28]:
```suggestion
* Archive Node: At least 2TB is required to store all historical state.
```

File: book/installation/installation.md [hunk 3]
```diff
@@ -18,19 +28,21 @@ There are multiple types of disks to sync Reth, with varying size requirements,
 * Archive Node: At least 2.5TB is required to store 
 * Full Node: TBD
 
-The time to sync also varies depending on the node type. NVMe drives are recommended for the best performance, however they can get expensive. SSDs are a good alternative, and HDDs are the cheapest option, but they will take the longest to sync.
+NVMe drives are recommended for the best performance, with SSDs being a cheaper alternative. HDDs are the cheapest option, but they will take the longest to sync, and are not recommended.
```
Comment (gakonst) [inline book/installation/installation.md:31]:
```suggestion
NVMe drives are recommended for the best performance, with SSDs being a cheaper alternative. HDDs are the cheapest option, but they will take the longest to sync, and are not recommended. At the time of writing, syncing an Ethereum mainnet node to block 17.4M on NVMe drives takes about 50 hours, while on a GCP "Persistent SSD" it takes ~5 days.
```

File: book/jsonrpc/admin.md [hunk 4]
```diff
@@ -1 +1,136 @@
...
+                "network": 1
+            }
+        }
+    }
+}
+```
+
+## `admin_peerEvents`, `admin_peerEvents_unsubscribe`
+
+<!-- TODO: This seems to be unimplemented, so it is not really known what the events look like !-->
```
Comment (gakonst) [inline book/jsonrpc/admin.md:116]: https://github.com/paradigmxyz/reth/blob/6ef3e12fac14a422c6a05d771126e47de8efee1c/crates/rpc/rpc/src/admin.rs#L54 yeah i thnk lets remove this

File: book/jsonrpc/intro.md [hunk 5]
```diff
@@ -1,19 +1,111 @@
...
+| [`eth`](./eth.md)       | The `eth` API allows you to interact with Ethereum.                                                    | Maybe     |
+| [`web3`](./web3.md)     | The `web3` API provides utility functions for the web3 client.                                         | No        |
+| [`net`](./net.md)       | The `net` API provides access to network information of the node.                                      | No        |
+| [`txpool`](./txpool.md) | The `txpool` API allows you to inspect the transaction pool.                                           | No        |
+| [`debug`](./debug.md)   | The `debug` API provides several methods to inspect the Ethereum state, including Geth-style traces.   | No        |
+| [`trace`](./trace.md)   | The `trace` API provides several methods to inspect the Ethereum state, including Parity-style traces. | No        |
+| [`admin`](./admin.md)   | The `admin` API allows you to configure your node.                                                     | **Yes**   |
+| [`rpc`](./rpc.md)       | The `rpc` API provides information about the RPC server and its modules.                               | No        |
```
Comment (gakonst) [inline book/jsonrpc/intro.md:24]: love it

File: book/jsonrpc/trace.md [hunk 6]
```diff
@@ -0,0 +1,183 @@
+# `trace` Namespace
+
+<!-- TODO: We should probably document the format of the traces themselves, OE does not do that !-->
```
Comment (gakonst) [inline book/jsonrpc/trace.md:3]: :+1: :+1: agreed - cc @mattsse

**Review by gakonst — APPROVED** [review]
great start so far - odds we can be ready with this by Monday? missing book chapters + empty sections on config. i think we can remove the setting up your local testnet, or can point to some commands around running with the auto-sealer. cc @Rjected if you have any good direction on "how do i run my node locally and just hack around in dev mode" for us to add to this section https://paradigmxyz.github.io/reth/run/local_testnet.html

File: book/jsonrpc/trace.md (hunk 6 above)
Comment (mattsse) [reply book/jsonrpc/trace.md:3]: by format, you mean the return type and all the fields? what they represent?

File: book/jsonrpc/admin.md (hunk 4 above)
Comment (mattsse) [reply book/jsonrpc/admin.md:116]: we can just keep it imo, doesn't really matter if we return implemented manually or let jsonrpsee return method not found error.

File: book/jsonrpc/trace.md (hunk 6 above)
Comment (onbjerg) [reply book/jsonrpc/trace.md:3]: yes, the structure is largely undocumented on the return types.

**Comment by onbjerg** [comment]
@gakonst I'll do the config in a separate PR.

There is also an outstanding task on updating all of the `--help` outputs on the page, but I'm thinking I'll just open a good first issue for that?

**Comment by gakonst** [comment]
Yes sounds good.

**Review by gakonst — APPROVED** [review]
Ship it

**Pull request merged**